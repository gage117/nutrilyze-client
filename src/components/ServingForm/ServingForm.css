.serving-form {
  width: 29rem;
  min-height: 46rem;
  background-color: #0c7b93;
  border-radius: .75rem;
  margin: 0 0 2.5rem;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
}

.sf__h2 {
  width: 29rem;
  padding: .25rem 0 0 1rem;
  margin: 0;
  font-size: 4rem;
  color: #40ceb8;
  border-radius: .75rem .75rem 0 0;
  background-color: #27496d;
}

.sf__p-container {
  width: 29rem;
  margin: 1rem;
}

.sf__p {
  margin: 0;
  padding: 0 0 0 1rem;
  font-size: 1.4rem;
  width: 27rem;
  text-align: justify;
}

.sf__input-container {
  background-color: #27496d;
  padding: 1rem 1.75rem;
  width: 24.5rem;
  border-radius: .75rem;
}

.sf__input-container > * {
  display: flex;
  justify-content: space-between;
}

#unit-of-measure {
  width: 14rem;
  background-color: whitesmoke;
}

.sf__text-input {
  width: 10rem;
  border-radius: .3rem;
  background-color: lightgray;
}

.sf__input-small {
  width: 5rem;
  border-radius: .3rem;
  background-color: lightgray;
}

.sf__block {
  display: block;
}

.sf__label {
  font-weight: bold;
  margin: .5rem;
  font-size: 1.6rem;
}

.sf__serving-size-label {
  display: block;
  margin: 0;
  padding: 0 0 0 .5rem;
}

.sf__serving-size-label > input {
  width: 3.5rem;
}

.sf__apply-to-label {
  margin: 1rem 1rem 0 1rem;
}

.sf__apply-to-span {
  display: inline-block;
  margin-bottom: .5rem;
}

.sf__autocomplete {
  display: inline-block;
  position: relative;
}

.sf__serving-size-note {
  font-weight: normal;
  padding-left: .5rem;
  display: block;
  font-size: 1.1rem;
}

.sf__submit-button-container {
  margin: 1.5rem 0;
  height: 3rem;
  width: 20.5rem;
  display: flex;
  justify-content: space-around;
  align-items: center;
}

.sf__submit-button {
  text-decoration: none;
  height: 3rem;
  font-size: 1.8rem;
  padding: .3rem .6rem;
  margin: 0 .3rem;
  border: 1px solid #142850;
  border-radius: .3rem;
  color: whitesmoke;
  background-color: #142850;
}

.sf__submit-button:hover {
  cursor: pointer;
  background-color: #40ceb8;
}

.sf__store-item-label {
  background-color: #27496d;
  display: flex;
  justify-content: space-between;
  align-items: center;
  height: 3rem;
  padding: 0 .5rem;
  margin: 0;
  border-radius: .5rem;
}

.sf__store-item-label > span {
  font-weight: normal;
}

/* ==Error Element== */
.sf__error {
  font-size: 1.5rem;
  font-weight: bold;
  text-align: center;
  color: #ECFFB6;
}

@media screen and (min-width: 885px) {
  .serving-form {
    margin: 0;
  }
}