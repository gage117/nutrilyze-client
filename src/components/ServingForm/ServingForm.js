import React, { Component } from 'react'
import './ServingForm.css'

class ServingForm extends Component {

  /** Multiple users */
  /** Returns the checkbox and label JSX of a user */
  // generateUserChoices = (user) => {
  //   return (
  //     <div key={user.id}>
  //       <input type='checkbox' className='user-checkbox' id={user.name} name={user.name} />
  //       <label htmlFor={user.name}>{user.name}</label>
  //     </div>
  //   )
  // }

  handleSubmitForm = (event) => {
    event.preventDefault();

    /** assigns the nutrition values from the input fields */
    const nutritionValues = {
      calories: event.target['calories'].value,
      protein: event.target['protein'].value,
      carbs: event.target['carbs'].value,
      sugar: event.target['sugar'].value,
      fiber: event.target['fiber'].value,
      fat: event.target['fat'].value,
      sodium: event.target['sodium'].value,
    }
    

    /** Iterates over the users and checks if that user's checkbox is checked
     *  -if so: It adds it to the usersToUpdate array to be passed into the updateUserNutrition function
     */
    const usersToUpdate = [this.props.users[0]];
    /** Uncomment this for loop if you want more than one user */
    // for (let i = 0; i < this.props.users.length; i++) {
    //   if (event.target[this.props.users[i].name].checked) {
    //     usersToUpdate.push(this.props.users[i])
    //   }
    // }
    

    /**
     * Checks if the "Store Item" checkbox is checked
     * If so:
     *  -it makes sure "Serving Size" has a value to store with the food
     *  -it runs the storeFood function passed down by App.js
     *  -it runs the updateUserNutrition function passed down by App.js
     * If not:
     *  -it runs only the updateUserNutrition, and doesn't require a serving size
     */
    switch(event.target['store-item'].checked) {
      case (true):
        if (!event.target['serving-size'].value) {
          alert('Serving Size requires a value if "Store Item" is checked');
        } else {
          const foodToStore = {
            name: event.target['name'].value,
            ...nutritionValues,
            serving_size: event.target['serving-size'].value,
            unit_of_measure: event.target['unit-of-measure'].value
          }
    
          this.props.handlers.storeFood(foodToStore)
          this.props.handlers.updateUserNutrition(nutritionValues, usersToUpdate)
        }
        break;
      default:
        this.props.handlers.updateUserNutrition(nutritionValues, usersToUpdate)
        break
    }
  }

  /** Paste this back in after the serving size label to have Apply-to Checkboxes for multiple users */
  // <label id='apply-to' className='apply-to-label block'>
  //   <span className='apply-to-span'>Apply To: </span>
  //     {/** Maps over users and generates a checkbox for each user */}
  //     {this.props.users.map((user) => {
  //       return this.generateUserChoices(user)
  //     })}
  // </label>

  render() {
    return (
      <form className='serving-form' onSubmit={this.handleSubmitForm} autoComplete='off'>
        <h2 className='sf__h2'>Track Food</h2>
        <div className='sf__p-container'>
          <p className='sf__p'>Add nutrition info of a food item to your current nutrition and/or store the item for later</p>
        </div>
        <div className='sf__input-container'>
          <label htmlFor='name' className='sf__label'>
            <span>Name: </span>
            <div className='sf__autocomplete'>
              <input id='name' className='sf__text-input' name='name' type='text' placeholder='Broccoli' required />
            </div>
          </label>
          <label htmlFor='calories' className='sf__label'>
            <span>Calories: </span>
            <input id='calories' className='sf__input-small' name='calories' type='text' placeholder='340' required />
          </label>
          <label htmlFor='protein' className='sf__label'>
            <span>Protein: </span>
            <input id='protein' type='text' className='sf__input-small' name='protein' placeholder='12g' required />
          </label>
          <label htmlFor='carbs' className='sf__label'>
            <span>Carbs: </span>
            <input id='carbs' type='text' className='sf__input-small' name='carbs' placeholder='17g' required />
          </label>
          <label htmlFor='sugar' className='sf__label'>
            <span>Sugar: </span>
            <input id='sugar' type='text' className='sf__input-small' name='sugar' placeholder='10g' required />
          </label>
          <label htmlFor='fiber' className='sf__label'>
            <span>Fiber: </span>
            <input id='fiber' type='text' className='sf__input-small' name='fiber' placeholder='8g' required />
          </label>
          <label htmlFor='fat' className='sf__label'>
            <span>Fat: </span>
            <input id='fat' type='text' className='sf__input-small' name='fat' placeholder='13g' required />
          </label>
          <label htmlFor='sodium' className='sf__label'>
            <span>Sodium: </span>
            <input id='sodium' type='text' className='sf__input-small' name='sodium' placeholder='10mg' required />
          </label>
          <label htmlFor='serving-size' className='sf__label sf__block sf__serving-size-label'>
            <span className='sf__block'>Serving Size: </span>
            <input id='serving-size' type='text' className='sf__text-input' name='serving-size' placeholder='2' />
            <select id='unit-of-measure'>
              <optgroup label='Weight'>
                <option value='oz'>Ounces{'(oz)'}</option>
                <option value='g'>Grams{'(g)'}</option>
              </optgroup>
              <optgroup label='Volume'>
                <option value='cups'>Cups</option>
                <option value='tspn'>Teaspoon{'(tspn)'}</option>
                <option value='tblspn'>Tablespoon{'(tblspn)'}</option>
              </optgroup>
              <optgroup label='Other'>
                <option value='container'>Container</option>
              </optgroup>
            </select>
            <span className='sf__serving-size-note'>Only required if "Store Item" is checked</span>
          </label>
        </div>
        <div className='sf__submit-button-container'>
          <button className='sf__submit-button' type='submit'>Submit</button>
          <label className='sf__label sf__store-item-label'>
            <span htmlFor='store-item'>Store Item: </span> 
            <input id='store-item' name='store-item' type='checkbox' />
          </label>
        </div>
      </form>
    )
  }
}

export default ServingForm
