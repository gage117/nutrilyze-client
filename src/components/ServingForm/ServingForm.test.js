import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import ServingForm from './ServingForm'

describe('ServingForm', () => {
  const users = [
    {
      id: 1,
      name: 'Gage',
      calories: 666,
      carbs: 40,
      protein: 20,
      fat: 30,
      sodium: 20
    },
    {
      id: 2,
      name: 'Alanna',
      calories: 0,
      carbs: 0,
      protein: 0,
      fat: 0,
      sodium: 0
    }
  ]
  
  it('renders without crashing', () => {
  
    const div = document.createElement('div')
    ReactDOM.render(<ServingForm users={users} />, div)
    ReactDOM.unmountComponentAtNode(div)
  })
  
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<ServingForm users={users} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})