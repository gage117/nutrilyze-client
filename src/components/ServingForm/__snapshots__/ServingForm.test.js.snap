// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServingForm renders the UI as expected 1`] = `
<form
  autoComplete="off"
  className="serving-form"
  onSubmit={[Function]}
>
  <h2
    className="sf__h2"
  >
    Track Food
  </h2>
  <div
    className="sf__p-container"
  >
    <p
      className="sf__p"
    >
      Add nutrition info of a food item to your current nutrition and/or store the item for later
    </p>
  </div>
  <div
    className="sf__input-container"
  >
    <label
      className="sf__label"
      htmlFor="name"
    >
      <span>
        Name: 
      </span>
      <div
        className="sf__autocomplete"
      >
        <input
          className="sf__text-input"
          id="name"
          name="name"
          placeholder="Broccoli"
          required={true}
          type="text"
        />
      </div>
    </label>
    <label
      className="sf__label"
      htmlFor="calories"
    >
      <span>
        Calories: 
      </span>
      <input
        className="sf__input-small"
        id="calories"
        name="calories"
        placeholder="340"
        required={true}
        type="text"
      />
    </label>
    <label
      className="sf__label"
      htmlFor="protein"
    >
      <span>
        Protein: 
      </span>
      <input
        className="sf__input-small"
        id="protein"
        name="protein"
        placeholder="12g"
        required={true}
        type="text"
      />
    </label>
    <label
      className="sf__label"
      htmlFor="carbs"
    >
      <span>
        Carbs: 
      </span>
      <input
        className="sf__input-small"
        id="carbs"
        name="carbs"
        placeholder="17g"
        required={true}
        type="text"
      />
    </label>
    <label
      className="sf__label"
      htmlFor="sugar"
    >
      <span>
        Sugar: 
      </span>
      <input
        className="sf__input-small"
        id="sugar"
        name="sugar"
        placeholder="10g"
        required={true}
        type="text"
      />
    </label>
    <label
      className="sf__label"
      htmlFor="fiber"
    >
      <span>
        Fiber: 
      </span>
      <input
        className="sf__input-small"
        id="fiber"
        name="fiber"
        placeholder="8g"
        required={true}
        type="text"
      />
    </label>
    <label
      className="sf__label"
      htmlFor="fat"
    >
      <span>
        Fat: 
      </span>
      <input
        className="sf__input-small"
        id="fat"
        name="fat"
        placeholder="13g"
        required={true}
        type="text"
      />
    </label>
    <label
      className="sf__label"
      htmlFor="sodium"
    >
      <span>
        Sodium: 
      </span>
      <input
        className="sf__input-small"
        id="sodium"
        name="sodium"
        placeholder="10mg"
        required={true}
        type="text"
      />
    </label>
    <label
      className="sf__label sf__block sf__serving-size-label"
      htmlFor="serving-size"
    >
      <span
        className="sf__block"
      >
        Serving Size: 
      </span>
      <input
        className="sf__text-input"
        id="serving-size"
        name="serving-size"
        placeholder="2"
        type="text"
      />
      <select
        id="unit-of-measure"
      >
        <optgroup
          label="Weight"
        >
          <option
            value="oz"
          >
            Ounces
            (oz)
          </option>
          <option
            value="g"
          >
            Grams
            (g)
          </option>
        </optgroup>
        <optgroup
          label="Volume"
        >
          <option
            value="cups"
          >
            Cups
          </option>
          <option
            value="tspn"
          >
            Teaspoon
            (tspn)
          </option>
          <option
            value="tblspn"
          >
            Tablespoon
            (tblspn)
          </option>
        </optgroup>
        <optgroup
          label="Other"
        >
          <option
            value="container"
          >
            Container
          </option>
        </optgroup>
      </select>
      <span
        className="sf__serving-size-note"
      >
        Only required if "Store Item" is checked
      </span>
    </label>
  </div>
  <div
    className="sf__submit-button-container"
  >
    <button
      className="sf__submit-button"
      type="submit"
    >
      Submit
    </button>
    <label
      className="sf__label sf__store-item-label"
    >
      <span
        htmlFor="store-item"
      >
        Store Item: 
      </span>
      <input
        id="store-item"
        name="store-item"
        type="checkbox"
      />
    </label>
  </div>
</form>
`;
